### PR TITLE
GH-2939: Update PULL_REQUEST_TEMPLATE

### LINKED ISSUE #2939: Update PULL_REQUEST_TEMPLATE
### Describe the enhancement requested

I'd like to update the PR template to follow what Apache Arrow does with a slight modification: https://github.com/apache/arrow/blob/main/.github/pull_request_template.md

### Component(s)

Build

### PR DESCRIPTION
**Pull request by wgtmac**
### Rationale for this change

The original PR template is a little bit out of date, especially after moving from JIRA to Github issue.

### What changes are included in this PR?

Update PULL_REQUEST_TEMPLATE to remove unnecessary check boxes and follow what Apache Arrow does. 

### Are these changes tested?

This is not a code change.

### Are there any user-facing changes?

No.

Closes #2939

### CODE REVIEW

**Comment by wgtmac** [comment]
@rok @gszadovszky @Fokko @julienledem Would you mind taking a look?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,33 +1,70 @@
-Make sure you have checked _all_ steps below.
 
-### Issue
+<!--
+Thanks for opening a pull request!
+If this is your first pull request you can find detailed information on how
```
Comment (Fokko) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: If you're new to Parquet-Java, information on how to contribute can be found here: https://parquet.apache.org/docs/contribution-guidelines/contributing

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,33 +1,70 @@
...
+Thanks for opening a pull request!
+If this is your first pull request you can find detailed information on how 
+to contribute here: [Contributing Overview](https://parquet.apache.org/docs/contribution-guidelines/contributing)
 
-- [ ] My PR addresses the following [GitHub issues](https://github.com/apache/parquet-java/issues) and references
-  them in the PR title. For example, "GH-1234: My Parquet PR"
-    - https://github.com/apache/parquet-java/issues/1234
-    - In case you are adding a dependency, check if the license complies with
-      the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
+If this is not a minor PR, could you open an issue for this pull request on GitHub? https://github.com/apache/parquet-java/issues/new/choose
```
Comment (Fokko) [inline .github/PULL_REQUEST_TEMPLATE.md:7]:
```suggestion
when contributing large features, open an issue for this pull request on GitHub:  https://github.com/apache/parquet-java/issues/new/choose
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,33 +1,70 @@
...
 
-### Tests
+Opening GitHub issues ahead of time contributes to the [Openness](http://theapacheway.com/open/#:~:text=Openness%20allows%20new%20users%20the,must%20happen%20in%20the%20open.) of the Apache Parquet-Java project.
```
Comment (Fokko) [inline .github/PULL_REQUEST_TEMPLATE.md:9]:
```suggestion
Opening GitHub issues ahead of time ensure openness of the Apache Parquet-Java project, read more about this here: http://theapacheway.com/open/#:~:text=Openness%20allows%20new%20users%20the,must%20happen%20in%20the%20open.
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,33 +1,70 @@
...
 
-- [ ] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:
+Then could you also rename the pull request title in the following format?
```
Comment (Fokko) [inline .github/PULL_REQUEST_TEMPLATE.md:11]:
```suggestion
Please format pull request title in the following format:
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,33 +1,70 @@
...
+the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
+-->
+<!-- **This PR adds new dependency.** -->
+
+<!--
+If there are any breaking changes to public APIs, please uncomment the line below and explain which changes are breaking.
+-->
+<!-- **This PR includes breaking changes to public APIs.** -->
+
+<!--
```
Comment (Fokko) [inline .github/PULL_REQUEST_TEMPLATE.md:61]: I might want to leave the critical fix out here. I dislike long PR templates, and I find that people don't really read them :'( Less is more I would say, or we should link to existing content (for example, link to the ASF Security page: https://www.apache.org/security/).

**Review by Fokko — COMMENTED** [review]
Thanks for picking this up @wgtmac 

I left a few comments. My personal preference is that these PR templates should be as short and concise as possible. I think that our old template is already a bit on the long side. My feeling is that people don't read them when they become too long. That's why I'm a bit of a nit-pick on the wording, sorry for that. I also suggested removing the markdown from the comments, since they won't render anyway.

Let me know what you think!

**Review by gszadovszky — COMMENTED** [review]
Thanks @wgtmac for doing this. I agree with @Fokko's comments, otherwise LGTM.

**Comment by wgtmac** [comment]
Thanks for the suggestion! I've simplified the wording and removed unimportant comments. PTAL. @Fokko @gszadovszky

**Comment by wgtmac** [comment]
@Fokko Gentle ping :)

**Review by Fokko — APPROVED** [review]
Looks good @wgtmac Thanks for fixing this 👍

**Comment by wgtmac** [comment]
Thanks @Fokko for merging!

**Pull request merged**